### ISSUE TITLE
Insert size=small annotations for subset of tests [phab:D920]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1231]
_Created by **djrenren** at 2016-06-06 20:47:55_: 

Bazel has optional test size annotations that describe resource usage and timeouts. We should probably use these (if only to get rid of that annoying warning when you run build). I've added some annotations so we can talk about style (where to put the size=small annotation) before I go through and add it to all the tests.

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-06 20:49:08_:

Not sure I like putting the size=small after the deps. Just figured we should come up with a convention before I edit like 50 files incorrectly

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-06 23:54:21_:

  - Add the size=small annotations where missing

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-06 23:56:06_:

  - Move the size attribute to the proper place
Missed the original set when updating the others

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-06 23:58:07_:

The solution I came up with marks all our go tests as small which is not ideal. There are also two more tests that are marked large but bazel complains about. As per the docs, 
```
Unittests are considered "small", integration tests "medium", and end-to-end tests "large" or "enormous". Bazel uses the size only to determine a default timeout.
```
This may have been a waste of time if we really want to adhere to this naming scheme. I could always replace the tests with timeouts as well in order to suppress the warning. Otherwise we can just forget the whole thing and ignore the warnings. Thoughts?

**Comment by kythe** [comment]
_**zarko** wrote at 2016-06-07 00:27:21_:

If you want to annotate the Go tests, I'd suggest doing so in a separate diff as you'll need to change the Skylark code (as you've partially done in /go.bzl, but in a final version size will probably need to be passed in to the macro).

The C++ indexer test formerly marked as "large" may now be classifiable as "small"; the testing infrastructure was improved, but evidently the annotation wasn't. The second (plan_ts_repo) should be kept the same, since it's large when the (deprecated) TypeScript tests are run.

It's still helpful to have the size annotations on other tests because we'll be able to catch performance regressions.

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-07 17:13:36_:

  - Remove blanket go size annotations
  - Update C++ indexer test size from large to small

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2016-06-07 17:51:07_

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-06-07 17:53:14_:

Please fix comment formatting

**Comment by kythe** [comment]
_**schroederc** required changes at 2016-06-07 17:53:14_

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-06-07 18:15:49_:

Reverted comment styling

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2016-06-07 18:16:51_

**Comment by kythe** [comment]
_**djrenren** closed via commit af080d4044cffe5d3ca860dad8463b814a145ae0 at 2016-06-07 19:40:21_
